website/content/tutorials/3-constraints-expect/_index.md

```diff
@@ -0,0 +1,1511 @@
+                    <formal-name>Vendor Name</formal-name>
+                    <description>The registered company name of the vendor.</description>
+                  </define-field>
+                  <define-field name="address" min-occurs="1" max-occurs="1">
+                    <formal-name>Vendor Address</formal-name>
+                    <description>The physical address of an office location for the vendor.</description>
+                  </define-field>
+                  <define-field name="website" as-type="uri" min-occurs="1" max-occurs="1">
+                    <formal-name>Vendor Website</formal-name>
+                    <description>A public website made by the vendor documenting their parts as used in the computer.</description>
+                  </define-field>
+                </model>
+              </define-assembly>
+              <define-field name="product-name" as-type="string" min-occurs="1" max-occurs="1">
+                <formal-name>Product Name</formal-name>
+                <description>The product name from the vendor of the computer part.</description>
+              </define-field>
+              <define-field name="type" as-type="string" min-occurs="1" max-occurs="1">
+                <formal-name>Expansion Card Type</formal-name>
+                <description>The type of expansion card on a motherboard of a computer, such as <code>pci</code> (PCI, e.g. Peripheral Component Interconnect), <code>pcie</code> (PCI Express), or an alternative.</description>
+              </define-field>
+            </model>
+          </define-assembly>
+        </model>
+        <constraint>
+          <expect id="memory-same-byte-size" level="CRITICAL" target="." test="if (count(./memory/byte-size) > 0) then (sum(./memory/byte-size) mod ./memory/byte-size[1]) = 0 else (sum(./memory/size) mod ./memory/size[1]) = 0">
+              <message>All memory modules SHOULD be the same size or byte-size for a computer.</message>
+          </expect>
+        </constraint>
+      </define-assembly>
+    </model>
+  </define-assembly>
+</METASCHEMA>
+```
+
+With our modifications to the model with the `choice` construct, we enabled our Metaschema-enabled tooling to support a field of `byte-size` or `size` where previously only the former was valid. For `size`, we added important constraints to its flag combinations to identify which unit (bits or bytes), unit prefix, unit system, and base (digital or binary) of the size's count. With the `byte-size` field, we add an `expect` constraint with a specific message to only present warning if information if model and tool developers use this deprecated field in document instances. Otherwise it can be ignored. Moreover, we refactored the previous size count constraints to check the logical constraint whether the developers use the deprecated field or the replacement field with the same logic using an `if` `else` expression.
```

Comment:
```suggestion
With our modifications to the model with the `choice` construct, we enabled our Metaschema-enabled tooling to support a field of `byte-size` or `size` where previously only the former was valid. For `size`, we added important constraints to its flag combinations to identify which unit (bits or bytes), unit prefix, unit system, and base (digital or binary) of the size's count. With the `byte-size` field, we add an `expect` constraint with a specific message to only present a warning if developers use this deprecated field in document instances; otherwise, the field can be ignored. Moreover, we refactored the previous size count constraints to check the logical constraint regardless of when the deprecated field or the replacement field is used. We support this with the same logic using an `if` `else` expression.
```